backend/app/services/device/version_checker.py

```
@@ -0,0 +1,115 @@
+            session = traced_session()
+            logger.debug(f"Fetching version from GitHub: {api_url}")
+            response = session.get(api_url, headers=headers, timeout=self.API_TIMEOUT)
+            response.raise_for_status()
+
+            data = response.json()
+            tag_name = data.get("tag_name", "")
+            latest_version = tag_name.lstrip("v")
+            logger.info(f"Fetched version from GitHub ({api_url}): {latest_version}")
+            # For backend, we don't need download URL, just version
+            return VersionInfo(version=latest_version, download_url="")
+        except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)